owtf/api/handlers/auth.py

```diff
@@ -0,0 +1,230 @@
+"""
+owtf.api.handlers.auth
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+"""
+from owtf.api.handlers.base import APIRequestHandler
+from owtf.lib.exceptions import APIError
+from owtf.models.user import User
+from datetime import datetime, timedelta
+import tornado
+import bcrypt
+import json
+import base64
+import uuid
+import jwt
+import re
+
+JWT_SECRET = 'my_secret_key'
+JWT_ALGORITHM = 'HS256'
+JWT_EXP_DELTA_SECONDS = 60 * 60 * 24
+
+
+class LogInHandler(APIRequestHandler):
+    """LogIn using the correct credentials (email, password). After successfull login a JWT Token is generated"""
+
+    SUPPORTED_METHODS = ["POST"]
+
+    def post(self):
+        """
+            **Example request**:
+
+            .. sourcecode:: http
+
+            POST /api/v1/login/ HTTP/1.1
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            **Login successful response**;
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok",
+                    "jwt-token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxOCwiZXhwIjoxNjE1NjU0OTk1fQ.wSMr5laDNVTxuTTJI4FWssnZa2xYL4PS8llDLDB4eyY"
+                }
+            }
+
+            **Login failed response**;
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "fail",
+                    "message": "Invalid login credentials"
+                }
+            }
+
+        """
+        body_data = json.loads(self.request.body.decode('utf-8'))
+        email = body_data['email']
+        password = body_data['password']
+        user = User.find_by_email(self.session, email)[0]
+
+        if user and user.password and bcrypt.hashpw(password.encode('utf-8'), user.password.encode('utf-8')) == user.password.encode('utf-8'):
+            self.set_current_user(email)
+            payload = {
+                'user_id': user.id,
+                'exp': datetime.utcnow() + timedelta(seconds=JWT_EXP_DELTA_SECONDS)
+            }
+            jwt_token = jwt.encode(payload, JWT_SECRET, JWT_ALGORITHM)
+            data = {
+                "status": "ok",
+                "jwt-token": jwt_token.decode('utf-8')
+            }
+            self.success(data)
+
+        else:
+            error_msg = "Invalid login credentials"
+            data = {
+                "status": "fail",
+                "message": error_msg
+            }
+            self.success(data)
+
+    def set_current_user(self, user):
```

Comment:
instead of setting a cookie, is it possible to use `Authorization: Bearer {}` everywhere - this basically mitigates any CSRF attack at all and makes code simpler by not having to manage cookies.

Reply:
Removed this part as it is no longer required as I had planned to store JWT token in the client-side and sending it in Authorization header as suggested.